docs/reference/architectures.rst

```diff
@@ -0,0 +1,183 @@
+Architectures
+=============
+
+snapcraft.yaml
+--------------
+
+The :doc:`architectures how-to</howto/architectures>` provides examples of how
+to use the ``architectures`` keyword.
+
+core22
+^^^^^^
+
+The root keyword ``architectures`` defines a list of ``build-on->build-for``
+pairs with the following syntax:
+
+.. code-block:: yaml
+
+  architectures:
+    - build-on: [<arch 1>, <arch 2>]
+      build-for: [<arch 1>]
+    - build-on: [<arch 3>]
+      build-for: [<arch 4>]
+      ...
+
+If the value is a single item, it can be simplified from a list to a scalar.
+For example:
+
+.. code-block:: yaml
+
+  architectures:
+    - build-on: [amd64]
+      build-for: [arm64]
+
+can be rewritten as:
+
+.. code-block:: yaml
+
+  architectures:
+    - build-on: amd64
+      build-for: arm64
+
+See :ref:`here<supported-architectures>` for a list of supported architectures.
+
+``build-for: all`` can be used for snaps that can run on all architectures,
```

Comment:
Is

```yaml
architectures:
  - all
```

valid, or do you have to declare a `build-on` if you want to `build-for` `all`?

Reply:
You have to declare a `build-on` in order to use `build-for: [all]`.  

I've made it more clear in the docs, added a link to the how-to, and filed an [issue](https://github.com/snapcore/snapcraft/issues/4361) to make the user experience better.